runner/src/pipeline.rs

```diff
@@ -83,21 +95,26 @@ impl LayerFunction {
     ) -> PyResult<PyObject> {
         async_python_run!(|py| {
             let args = if self.takes_input_arg {
-                PyTuple::new(py, [deepcopy(py, input.as_ref(py))?])
+                PyTuple::new(py, [deepcopy(py, input.into_bound_py_any(py)?)?])?
```

Comment:
if you rewrite `deepcopy` like I suggest below, you could just do `deepcopy(input.bind(py))`